webclient/webclient/src/main/java/io/helidon/webclient/NettyClientHandler.java

```diff
@@ -197,12 +197,13 @@ private boolean shouldResponseAutomaticallyClose(WebClientResponse clientRespons
     }
 
     private boolean noContentLength(WebClientResponseHeaders headers) {
-        return headers.first(Http.Header.CONTENT_LENGTH).isEmpty()
-                || headers.first(Http.Header.CONTENT_LENGTH).get().equals("0");
+        return headers.contentLength()
+                .map(value -> value == 0)
+                .orElse(true);
     }
 
     private boolean notChunked(WebClientResponseHeaders headers) {
-        return headers.first(Http.Header.TRANSFER_ENCODING).isEmpty();
+        return headers.transferEncoding().isEmpty();
```

Comment:
I think just checking for `isEmpty` is not sufficient. Transfer encoding can have the following values:
```
Transfer-Encoding: chunked
Transfer-Encoding: compress
Transfer-Encoding: deflate
Transfer-Encoding: gzip
Transfer-Encoding: identity
```